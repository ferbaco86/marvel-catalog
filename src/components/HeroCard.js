import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const imageSize = 'portrait_uncanny';

const Card = styled.div`
display: flex;
flex-direction: column;
align-items: center;
background-color: #202020;
flex: 1 0 15%;
margin: 2rem;`;

const Title = styled.h1`
color: white;
font-family: 'Roboto Condensed', sans-serif;
font-weight: 700;
text-align: center;`;

const SubTitle = styled.p`
color: white;`;

const Pic = styled.img`
border-bottom: 3px solid #ec1d24`;

const HeroCard = props => {
  const {
    name, image, extension, description,
  } = props;
  return (
    <Card>
      <Pic src={`${image}/${imageSize}.${extension}`} alt={`${name} Portrait`} />
      <Title>{name}</Title>
      <SubTitle>{description}</SubTitle>
    </Card>
  );
};

HeroCard.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  extension: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default HeroCard;
