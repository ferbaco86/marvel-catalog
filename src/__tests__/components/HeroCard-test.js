import { render, screen, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../reducers/index';
import HeroCard from '../../components/HeroCard';

const character = {
  name: 'Blade',
  image: 'www.mockurl.com',
  extension: 'jpg',
  url: 'www.mockurl2.com',
};

describe('Rendering component', () => {
  it('creates a HeroCard component', () => {
    act(() => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <HeroCard
              name={character.name}
              image={character.image}
              extension={character.extension}
              url={character.url}
            />
          </Provider>
        </BrowserRouter>,
      );
    });
    screen.getByText('Blade');
  });
});

describe('Display', () => {
  it('renders correctly', () => {
    const snap = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <HeroCard
            name={character.name}
            image={character.image}
            extension={character.extension}
            url={character.url}
          />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
