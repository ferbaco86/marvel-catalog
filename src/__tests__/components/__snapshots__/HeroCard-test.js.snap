// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders correctly 1`] = `
<div
  className="sc-bdfBwQ iEmcFJ"
>
  <a
    className="sc-hKgILt bWnvia"
    href="www.mockurl2.com"
  >
    <img
      alt="Blade Portrait"
      className="sc-dlfnbm ckVWtK"
      src="www.mockurl.com/portrait_uncanny.jpg"
    />
    <h1
      className="sc-gsTCUz kqqFgO"
    >
      Blade
    </h1>
  </a>
</div>
`;
